specs/signature-envelope-cose.md

```diff
@@ -239,6 +239,55 @@ The final signature envelope is a `COSE_Sign1_Tagged` object, consisting of Payl
+The differences between a `COSE Hash Envelope` and the signature envelope defined under [Signature Envelope](./signature-envelope-cose.md/#signature-envelope) are as follows:
+1. Instead of `targetArtifact` as shown under [COSE Payload](./signature-envelope-cose.md/#cose-payload), the payload of a `COSE Hash Envelope` is the hash of the original content bytes that is being signed. The hash algorithm MUST be deduced from the signing certificate's public key (see [Algorithm Selection](./signature-specification.md#algorithm-selection)). In other words, the digest value of [`targetArtifact.digest`](./signature-envelope-cose.md/#cose-payload) MUST be saved as the `payload` when `COSE Hash Envelope` is used.
+2. For `targetArtifact.mediaType`, `targetArtifact.size`, and `targetArtifact.artifactType`, they are mapped as new protected headers under `COSE Hash Envelope`. For `targetArtifact,annotations`, if present, each annotation MUST be stored as a single [`Extended attributes`](./signature-specification.md/#extended-attributes) under `COSE Hash Envelope`. See details below.
+3. New protected header parameters under `COSE Hash Envelope` only:
+    - **[`payload hash alg`](https://datatracker.ietf.org/doc/draft-ietf-cose-hash-envelope/05/)** (*int*): REQUIRED protected header payload hash alg (label `258`) specifies the hash algorithm used to produce the payload.
+    - **[`preimage content type`](https://datatracker.ietf.org/doc/draft-ietf-cose-hash-envelope/05/)** (*uint/tstr*): REQUIRED protected header preimage content type (label `259`) specifies the content type of the bytes that were hashed (preimage) to produce the payload, given as a content-format number (Section 12.3 of [RFC7252](https://datatracker.ietf.org/doc/html/rfc7252#section-12.3)) or as a media-type name optionally with parameters (Section 8.3 of [RFC9110](https://datatracker.ietf.org/doc/html/rfc9110#section-8.3)). This value MUST have the same value as [`targetArtifact.mediaType`](./signature-envelope-cose.md/#cose-payload).
+    - **[`payload location`](https://datatracker.ietf.org/doc/draft-ietf-cose-hash-envelope/05/)** (*tstr*): OPTIONAL protected header payload location (label `260`) is an identifier enabling retrieval of the original resource (preimage) identified by the payload.
+    - **`io.cncf.notary.preimageSize`** (*int*): REQUIRED protected header `io.cncf.notary.preimageSize` specifies the size of the preimage. This value MUST have the same value as [`targetArtifact.size`](./signature-envelope-cose.md/#cose-payload).
+    - **`io.cncf.notary.artifactType`** (*tstr*): OPTIONAL protected header `io.cncf.notary.artifactType` specifies the artifactType of the preimage. This value MUST have the same value as [`targetArtifact.artifactType`](./signature-envelope-cose.md/#cose-payload).
```

Comment:
Do we accept `uint` as defined in https://www.iana.org/assignments/core-parameters/core-parameters.xhtml#content-formats?